I can see some build problem in the Guerrilla Smoketest related this PR:
'''
[ 44%] Building CXX object common/thorhelper/CMakeFiles/thorhelper.dir/thorstats.cpp.o
In file included from /home/centos/smoketest/PR-18233/build/vcpkg_installed/x64-linux-dynamic/include/arrow/util/async_generator.h:32,
                 from /home/centos/smoketest/PR-18233/build/vcpkg_installed/x64-linux-dynamic/include/arrow/acero/options.h:34,
                 from /home/centos/smoketest/PR-18233/build/vcpkg_installed/x64-linux-dynamic/include/arrow/dataset/scanner.h:28,
                 from /home/centos/smoketest/PR-18233/build/vcpkg_installed/x64-linux-dynamic/include/arrow/dataset/file_base.h:31,
                 from /home/centos/smoketest/PR-18233/build/vcpkg_installed/x64-linux-dynamic/include/arrow/dataset/api.h:25,
                 from /home/centos/smoketest/PR-18233/HPCC-Platform/plugins/parquet/parquetembed.hpp:26,
                 from /home/centos/smoketest/PR-18233/HPCC-Platform/common/thorhelper/thorread.cpp:39:
/home/centos/smoketest/PR-18233/build/vcpkg_installed/x64-linux-dynamic/include/arrow/util/io_util.h:122:87: error: macro "DeleteFile" passed 2 arguments, but takes just 1
  122 | Result<bool> DeleteFile(const PlatformFilename& file_path, bool allow_not_found = true);
      |                                                                                       ^
In file included from /home/centos/smoketest/PR-18233/HPCC-Platform/system/jlib/jliball.hpp:22,
                 from /home/centos/smoketest/PR-18233/HPCC-Platform/common/thorhelper/thorread.cpp:17:
/home/centos/smoketest/PR-18233/HPCC-Platform/system/include/platform.h:349: note: macro "DeleteFile" defined here
  349 | #define DeleteFile(name)    (unlink(name)==0)
      | 
'''
Maybe a rebase issue? Some related changes in platform.h merged yesterday.
@ghalliday any thoughts?

--- a/cmake_modules/commonSetup.cmake
+++ b/cmake_modules/commonSetup.cmake
@@ -865,6 +865,11 @@ IF ("${COMMONSETUP_DONE}" STREQUAL "")
         endif(USE_C11_REGEX)
       endif(USE_BOOST_REGEX)
 
+      if(USE_PARQUET)
+        message(STATUS "Parquet enabled")
+        add_definitions(-D_USE_PARQUET)
+      endif()
+
       if(USE_OPENSSL AND USE_OPENSSLV3)
         message(STATUS "OPENSSL Version 3 or greater enabled")
         add_definitions (-D_USE_OPENSSL)

--- a/cmake_modules/options.cmake
+++ b/cmake_modules/options.cmake
@@ -9,6 +9,7 @@ option(DEVEL "Enable the building/inclusion of a Development component." OFF)
 option(CLIENTTOOLS_ONLY "Enable the building of Client Tools only." OFF)
 option(INCLUDE_PLUGINS "Enable the building of platform and all plugins for testing purposes" OFF)
 option(USE_CASSANDRA "Include the Cassandra plugin in the base package" ON)
+option(USE_PARQUET "Include the Parquet plugin in the base package" ON)
 option(PLUGIN "Enable building of a plugin" OFF)
 option(USE_SHLIBDEPS "Enable the use of dpkg-shlibdeps on ubuntu packaging" OFF)
 

--- a/cmake_modules/plugins.cmake
+++ b/cmake_modules/plugins.cmake
@@ -80,6 +80,10 @@ if (USE_MYSQL_REPOSITORY)
     set(VCPKG_MYSQLEMBED "${VCPKG_INCLUDE}")
 endif()
 
+if (USE_PARQUET)
+    set(VCPKG_PARQUETEMBED "${VCPKG_INCLUDE}")
+endif()
+
 #  vcpkg.json options  ---
 set(VCPKG_APR "${VCPKG_SUPPRESS}")
 if (USE_APR)

--- a/common/thorhelper/CMakeLists.txt
+++ b/common/thorhelper/CMakeLists.txt
@@ -112,6 +112,13 @@ include_directories (
          ${HPCC_SOURCE_DIR}/fs/dafsclient
     )
 
+if (USE_PARQUET)
+    include_directories (
+         ${HPCC_SOURCE_DIR}/esp/platform
+         ${HPCC_SOURCE_DIR}/plugins/parquet
+    )
+endif()
+
 ADD_DEFINITIONS( -DTHORHELPER_EXPORTS -D_USRDLL )
 
 HPCC_ADD_LIBRARY( thorhelper SHARED ${SRCS} )
@@ -125,6 +132,10 @@ target_link_libraries ( thorhelper
          deftype 
          dalibase
     )
+
+if (USE_PARQUET)
+    target_link_libraries( thorhelper parquetembed )
+endif()
 
 if (USE_NUMA)
  target_link_libraries ( thorhelper ${NUMA_LIBRARIES} )

--- a/common/thorhelper/thorread.cpp
+++ b/common/thorhelper/thorread.cpp
@@ -14,7 +14,6 @@
     See the License for the specific language governing permissions and
     limitations under the License.
 ############################################################################## */
-
 #include "jliball.hpp"
 
 #include "thorfile.hpp"
@@ -35,6 +34,10 @@
 #include "thorcommon.hpp"
 #include "csvsplitter.hpp"
 #include "thormeta.hpp"
+
+#ifdef _USE_PARQUET
+    #include "parquetembed.hpp"
+#endif
 
 //---------------------------------------------------------------------------------------------------------------------
 
@@ -1272,7 +1275,208 @@ protected:
     IDiskRowReader * activeReader = nullptr;
 };
 
+//---------------------------------------------------------------------------------------------------------------------
+#ifdef _USE_PARQUET
+class CParquetActivityContext : public IThorActivityContext
+{
+public:
+    CParquetActivityContext(bool _local, unsigned _numWorkers, unsigned _curWorker)
+    : local(_local), workers(_local ? 1 : _numWorkers), curWorker(_local ? 0 : _curWorker)
+    {
+        assertex(curWorker < workers);
+    }
 
+    virtual bool isLocal() const override { return local; };
+    virtual unsigned numSlaves() const override { return workers; };
+    virtual unsigned numStrands() const override { return 1; };
+    virtual unsigned querySlave() const override { return curWorker; };
+    virtual unsigned queryStrand() const override { return 0; }; // 0 based 0..numStrands-1
+protected:
+    unsigned workers;
+    unsigned curWorker;
+    bool local;
+};
+
+/*
+ * Base class for reading a Parquet local file
+ */
+class ParquetDiskRowReader : public CInterfaceOf<IDiskRowStream>, implements IDiskRowReader
+{
+public:
+    ParquetDiskRowReader(IDiskReadMapping * _mapping);
+    ~ParquetDiskRowReader();
+    IMPLEMENT_IINTERFACE_USING(CInterfaceOf<IDiskRowStream>)
+
+    virtual IDiskRowStream * queryAllocatedRowStream(IEngineRowAllocator * _outputAllocator) override;
+
+    virtual const void * nextRow() override;
+    virtual const void * nextRow(size32_t & resultSize) override;
+    virtual const void * nextRow(MemoryBufferBuilder & builder) override;
+    virtual bool getCursor(MemoryBuffer & cursor) override;
+    virtual void setCursor(MemoryBuffer & cursor) override;
+    virtual void stop() override;
+
+    virtual void clearInput() override;
+    virtual bool matches(const char * _format, bool _streamRemote, IDiskReadMapping * _mapping) override;
+
+// IDiskRowReader
+    virtual bool setInputFile(const char * localFilename, const char * logicalFilename, unsigned partNumber, offset_t baseOffset, const IPropertyTree * inputOptions, const FieldFilterArray & expectedFilter) override;
+    virtual bool setInputFile(const RemoteFilename & filename, const char * logicalFilename, unsigned partNumber, offset_t baseOffset, const IPropertyTree * inputOptions, const FieldFilterArray & expectedFilter) override;
+    virtual bool setInputFile(const CLogicalFileSlice & slice, const FieldFilterArray & expectedFilter, unsigned copy) override;
+
+protected:
+    IDiskReadMapping * mapping = nullptr;
+    Owned<IEngineRowAllocator> outputAllocator;
+
+    MemoryBuffer tempOutputBuffer;
+    MemoryBufferBuilder bufferBuilder;
+
+    parquetembed::ParquetReader * parquetFileReader = nullptr;
+    CParquetActivityContext * parquetActivityCtx = nullptr;
+    StringAttr format;
+    RecordTranslationMode translationMode;
+    bool eogPending = false;
+};
+
+ParquetDiskRowReader::ParquetDiskRowReader(IDiskReadMapping * _mapping)
+ : mapping(_mapping), bufferBuilder(tempOutputBuffer, 0), parquetActivityCtx(new CParquetActivityContext(true, 1, 0))
+{
+    translationMode = mapping->queryTranslationMode();
+}
+
+ParquetDiskRowReader::~ParquetDiskRowReader()
+{
+    if (parquetFileReader)
+    {
+        delete parquetFileReader;
+    }
+
+    if (parquetActivityCtx)
+    {
+        delete parquetActivityCtx;
+    }
+}
+
+IDiskRowStream * ParquetDiskRowReader::queryAllocatedRowStream(IEngineRowAllocator * _outputAllocator)
+{
+    outputAllocator.set(_outputAllocator);
+    return this;
+}
+
+// Returns rows to the engine for the next stage in the processing
+const void * ParquetDiskRowReader::nextRow()
+{
+    while (parquetFileReader->shouldRead())
+    {
+        parquetembed::TableColumns * table = nullptr;
+        auto index = parquetFileReader->next(table);
+
+        if (table && !table->empty())
+        {
+            parquetembed::ParquetRowBuilder pRowBuilder(table, index);
+
+            RtlDynamicRowBuilder rowBuilder(outputAllocator);
+            const RtlTypeInfo * typeInfo = outputAllocator->queryOutputMeta()->queryTypeInfo();
+            assertex(typeInfo);
+            RtlFieldStrInfo dummyField("<row>", NULL, typeInfo);
+            size32_t sizeRead = typeInfo->build(rowBuilder, 0, &dummyField, pRowBuilder);
+            roxiemem::OwnedConstRoxieRow next = rowBuilder.finalizeRowClear(sizeRead);
+            return next.getClear();
+        }
+    }
+    return eofRow;
+}
+
+// Returns temporary rows for filtering/counting etc.
+// Row is built in temporary buffer and reused.
+const void * ParquetDiskRowReader::nextRow(size32_t & resultSize)
+{
+    const void * next = nextRow(bufferBuilder);
+    resultSize = tempOutputBuffer.length();
+    return next;
+}
+
+// Returns rows to any output buffer in dafilesrv
+const void * ParquetDiskRowReader::nextRow(MemoryBufferBuilder & builder)
+{
+    while (parquetFileReader->shouldRead())
+    {
+        parquetembed::TableColumns * table = nullptr;
+        auto index = parquetFileReader->next(table);
+
+        if (table && !table->empty())
+        {
+            parquetembed::ParquetRowBuilder pRowBuilder(table, index);
+
+            const RtlTypeInfo * typeInfo = outputAllocator->queryOutputMeta()->queryTypeInfo();
+            assertex(typeInfo);
+            RtlFieldStrInfo dummyField("<row>", NULL, typeInfo);
+            size32_t resultSize = typeInfo->build(builder, 0, &dummyField, pRowBuilder);
+            const void * next = builder.getSelf();
+            builder.finishRow(resultSize);
+            return next;
+        }
+    }
+    return nullptr;
+}
+
+bool ParquetDiskRowReader::getCursor(MemoryBuffer & cursor)
+{
+    return false;
+}
+
+void ParquetDiskRowReader::setCursor(MemoryBuffer & cursor)
+{
+}
+
+void ParquetDiskRowReader::stop()
+{
+}
+
+void ParquetDiskRowReader::clearInput()
+{
+    eogPending = false;
+}
+
+bool ParquetDiskRowReader::matches(const char * _format, bool _streamRemote, IDiskReadMapping * _mapping)
+{
+    if (!strieq(format, "parquet"))
+        return false;
+    return true; // TO DO add additional check
+}
+
+bool ParquetDiskRowReader::setInputFile(const char * localFilename, const char * logicalFilename, unsigned partNumber, offset_t baseOffset, const IPropertyTree * inputOptions, const FieldFilterArray & expectedFilter)
+{
+    DBGLOG(0, "Opening File: %s", localFilename);
+    parquetFileReader = new parquetembed::ParquetReader("read", localFilename, 50000, nullptr, parquetActivityCtx);
+    auto st = parquetFileReader->processReadFile();
+    if (!st.ok())
+        throw MakeStringException(0, "%s: %s.", st.CodeAsString().c_str(), st.message().c_str());
+    return true;
+}
+
+bool ParquetDiskRowReader::setInputFile(const RemoteFilename & filename, const char * logicalFilename, unsigned partNumber, offset_t baseOffset, const IPropertyTree * inputOptions, const FieldFilterArray & expectedFilter)
+{
+    throwUnexpected();
+}
+
+bool ParquetDiskRowReader::setInputFile(const CLogicalFileSlice & slice, const FieldFilterArray & expectedFilter, unsigned copy)
+{
+    StringBuffer localPath;
+    slice.getURL(localPath, copy);
+
+    // getURL returns a filename in the format ./file/127.0.0.1/path/to/file/data.parquet
+    // To use it with the ParquetReader constructor we need just the file path
+    // Skip leading .
+    const char * slash = strstr(localPath, "/");
+    // Skip file
+    slash = strstr(slash+1, "/");
+    // Skip ip
+    slash = strstr(slash+1, "/");
+
+    return setInputFile(slash, slice.queryLogicalFilename(), slice.queryPartNumber(), slice.queryStartOffset(), slice.queryFileMeta(), expectedFilter);
+}
+#endif
 //---------------------------------------------------------------------------------------------------------------------
 
 
@@ -1507,6 +1711,10 @@ void RemoteDiskRowReader::stop()
 
 IDiskRowReader * doCreateLocalDiskReader(const char * format, IDiskReadMapping * _mapping)
 {
+#ifdef _USE_PARQUET
+    if (strieq(format, "parquet"))
+        return new ParquetDiskRowReader(_mapping);
+#endif
     if (strieq(format, "flat"))
         return new BinaryDiskRowReader(_mapping);
     if (strieq(format, "csv"))

--- a/plugins/parquet/CMakeLists.txt
+++ b/plugins/parquet/CMakeLists.txt
@@ -26,9 +26,9 @@
 project(parquetembed)
 message("CMAKE Version: ${CMAKE_VERSION}")
 
-if(PARQUETEMBED)
+if(PARQUETEMBED OR USE_PARQUET)
     ADD_PLUGIN(parquetembed)
-    if(MAKE_PARQUETEMBED)
+    if(MAKE_PARQUETEMBED OR USE_PARQUET)
         find_package(Arrow CONFIG REQUIRED)
         find_package(Parquet CONFIG REQUIRED)
         find_package(ArrowDataset CONFIG REQUIRED)
@@ -39,7 +39,6 @@ if(PARQUETEMBED)
         )
 
         INCLUDE_DIRECTORIES(
-            ${HPCC_SOURCE_DIR}/esp/platform
             ${HPCC_SOURCE_DIR}/system/include
             ${HPCC_SOURCE_DIR}/rtl/eclrtl
             ${HPCC_SOURCE_DIR}/rtl/include
@@ -49,6 +48,8 @@ if(PARQUETEMBED)
             ${HPCC_SOURCE_DIR}/roxie/roxiemem
         )
 
+        add_definitions(-D_USRDLL -DPARQUETEMBED_PLUGIN_EXPORTS)
+
         HPCC_ADD_LIBRARY(parquetembed SHARED ${SRCS})
 
         install(
@@ -57,56 +58,18 @@ if(PARQUETEMBED)
         )
 
         install(
-            FILES ${LIBARROW_LIB_REAL}
-            DESTINATION ${LIB_DIR} CALC_DEPS
-            PERMISSIONS OWNER_WRITE OWNER_READ OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE
-            COMPONENT Runtime)
-
-        install(
-            FILES ${LIBARROW_LIB} ${LIBARROW_LIB_ABI}
-            DESTINATION ${LIB_DIR} CALC_DEPS
-            COMPONENT Runtime)
-
-        install(
-            FILES ${LIBPARQUET_LIB_REAL}
-            DESTINATION ${LIB_DIR} CALC_DEPS
-            PERMISSIONS OWNER_WRITE OWNER_READ OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE
-            COMPONENT Runtime)
-
-        install(
-            FILES ${LIBPARQUET_LIB} ${LIBPARQUET_LIB_ABI}
-            DESTINATION ${LIB_DIR} CALC_DEPS
-            COMPONENT Runtime)
-
-        install(
-            FILES ${LIBARRORACERO_LIB_REAL}
-            DESTINATION ${LIB_DIR} CALC_DEPS
-            PERMISSIONS OWNER_WRITE OWNER_READ OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE
-            COMPONENT Runtime)
-
-        install(
-            FILES ${LIBARRORACERO_LIB} ${LIBARROWDATASET_LIB_ABI}
-            DESTINATION ${LIB_DIR} CALC_DEPS
-            COMPONENT Runtime)
-
-        install(
-            FILES ${LIBARROWDATASET_LIB_REAL}
-            DESTINATION ${LIB_DIR} CALC_DEPS
-            PERMISSIONS OWNER_WRITE OWNER_READ OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE
-            COMPONENT Runtime)
-
-        install(
-            FILES ${LIBARROWDATASET_LIB} ${LIBARROWDATASET_LIB_ABI}
-            DESTINATION ${LIB_DIR} CALC_DEPS
-            COMPONENT Runtime)
+            TARGETS parquetembed
+            DESTINATION ${LIB_DIR}
+            COMPONENT Runtime CALC_DEPS
+        )
 
         target_link_libraries(
             parquetembed
             eclrtl
             jlib
-            Arrow::arrow_shared
-            Parquet::parquet_shared
-            ArrowDataset::arrow_dataset_shared
+            "$<IF:$<BOOL:${ARROW_BUILD_STATIC}>,Arrow::arrow_static,Arrow::arrow_shared>"
+            "$<IF:$<BOOL:${ARROW_BUILD_STATIC}>,Parquet::parquet_static,Parquet::parquet_shared>"
+            "$<IF:$<BOOL:${ARROW_BUILD_STATIC}>,ArrowDataset::arrow_dataset_static,ArrowDataset::arrow_dataset_shared>"
         )
     endif()
 endif()

--- a/plugins/parquet/parquetembed.cpp
+++ b/plugins/parquet/parquetembed.cpp
@@ -31,7 +31,7 @@ static const NullFieldProcessor NULLFIELD(NULL);
  * @brief Takes a pointer to an ECLPluginDefinitionBlock and passes in all the important info
  * about the plugin.
  */
-extern "C" PARQUETEMBED_PLUGIN_API bool getECLPluginDefinition(ECLPluginDefinitionBlock *pb)
+extern "C" DECL_EXPORT bool getECLPluginDefinition(ECLPluginDefinitionBlock *pb)
 {
     if (pb->size == sizeof(ECLPluginDefinitionBlockEx))
     {
@@ -298,6 +298,7 @@ std::shared_ptr<parquet::arrow::RowGroupReader> ParquetReader::queryCurrentTable
         offset = tables;
     }
     failx("Failed getting RowGroupReader. Index %lli is out of bounds.", currTable);
+    return nullptr;
 }
 
 /**
@@ -981,6 +982,7 @@ __int64 getSigned(std::shared_ptr<ParquetArrayVisitor> &arrayVisitor, int index)
         default:
             failx("getSigned: Invalid size %i", arrayVisitor->size);
     }
+    return 0;
 }
 
 /**
@@ -1005,6 +1007,7 @@ unsigned __int64 getUnsigned(std::shared_ptr<ParquetArrayVisitor> &arrayVisitor,
         default:
             failx("getUnsigned: Invalid size %i", arrayVisitor->size);
     }
+    return 0;
 }
 
 /**
@@ -1027,6 +1030,7 @@ double getReal(std::shared_ptr<ParquetArrayVisitor> &arrayVisitor, int index)
         default:
             failx("getReal: Invalid size %i", arrayVisitor->size);
     }
+    return 0;
 }
 
 /**
@@ -1042,32 +1046,32 @@ std::string_view ParquetRowBuilder::getCurrView(const RtlFieldInfo *field)
     switch(arrayVisitor->type)
     {
         case BoolType:
-            tokenSerializer.serialize(arrayVisitor->boolArr->Value(currArrayIndex()), serialized);
+            serialized.append(arrayVisitor->boolArr->Value(currArrayIndex()));
             return serialized.str();
         case BinaryType:
             return arrayVisitor->binArr->GetView(currArrayIndex());
         case LargeBinaryType:
             return arrayVisitor->largeBinArr->GetView(currArrayIndex());
         case RealType:
-            tokenSerializer.serialize(getReal(arrayVisitor, currArrayIndex()), serialized);
+            serialized.append(getReal(arrayVisitor, currArrayIndex()));
             return serialized.str();
         case IntType:
-            tokenSerializer.serialize(getSigned(arrayVisitor, currArrayIndex()), serialized);
+            serialized.append(getSigned(arrayVisitor, currArrayIndex()));
             return serialized.str();
         case UIntType:
-            tokenSerializer.serialize(getUnsigned(arrayVisitor, currArrayIndex()), serialized);
+            serialized.append(getUnsigned(arrayVisitor, currArrayIndex()));
             return serialized.str();
         case DateType:
-            tokenSerializer.serialize(arrayVisitor->size == 32 ? (__int32) arrayVisitor->date32Arr->Value(currArrayIndex()) : (__int64) arrayVisitor->date64Arr->Value(currArrayIndex()), serialized);
+            serialized.append(arrayVisitor->size == 32 ? (__int32) arrayVisitor->date32Arr->Value(currArrayIndex()) : (__int64) arrayVisitor->date64Arr->Value(currArrayIndex()));
             return serialized.str();
         case TimestampType:
-            tokenSerializer.serialize((__int64) arrayVisitor->timestampArr->Value(currArrayIndex()), serialized);
+            serialized.append((__int64) arrayVisitor->timestampArr->Value(currArrayIndex()));
             return serialized.str();
         case TimeType:
-            tokenSerializer.serialize(arrayVisitor->size == 32 ? (__int32) arrayVisitor->time32Arr->Value(currArrayIndex()) : (__int64) arrayVisitor->time64Arr->Value(currArrayIndex()), serialized);
+            serialized.append(arrayVisitor->size == 32 ? (__int32) arrayVisitor->time32Arr->Value(currArrayIndex()) : (__int64) arrayVisitor->time64Arr->Value(currArrayIndex()));
             return serialized.str();
         case DurationType:
-            tokenSerializer.serialize((__int64) arrayVisitor->durationArr->Value(currArrayIndex()), serialized);
+            serialized.append((__int64) arrayVisitor->durationArr->Value(currArrayIndex()));
             return serialized.str();
         case StringType:
             return arrayVisitor->stringArr->GetView(currArrayIndex());
@@ -1078,6 +1082,7 @@ std::string_view ParquetRowBuilder::getCurrView(const RtlFieldInfo *field)
         default:
             failx("Unimplemented Parquet type for field with name %s.", field->name);
     }
+    return "";
 }
 
 /**
@@ -1108,11 +1113,8 @@ __int64 ParquetRowBuilder::getCurrIntValue(const RtlFieldInfo *field)
             return arrayVisitor->durationArr->Value(currArrayIndex());
         default:
         {
-            __int64 myint64 = 0;
             auto scalar = getCurrView(field);
-            std::string scalarStr(scalar.data(), scalar.size());
-            handleDeserializeOutcome(tokenDeserializer.deserialize(scalarStr.c_str(), myint64), "signed", scalarStr.c_str());
-            return myint64;
+            return rtlStrToInt8(scalar.size(), scalar.data());
         }
     }
 }
@@ -1145,11 +1147,8 @@ double ParquetRowBuilder::getCurrRealValue(const RtlFieldInfo *field)
             return arrayVisitor->durationArr->Value(currArrayIndex());
         default:
         {
-            double mydouble = 0.0;
             auto scalar = getCurrView(field);
-            std::string scalarStr(scalar.data(), scalar.size());
-            handleDeserializeOutcome(tokenDeserializer.deserialize(scalarStr.c_str(), mydouble), "real", scalarStr.c_str());
-            return mydouble;
+            return rtlStrToReal(scalar.size(), scalar.data());
         }
     }
 }

--- a/plugins/parquet/parquetembed.hpp
+++ b/plugins/parquet/parquetembed.hpp
@@ -38,7 +38,6 @@
 #include "hqlplugins.hpp"
 #include "eclrtl_imp.hpp"
 #include "eclhelper.hpp"
-#include "tokenserialization.hpp"
 #include "rtlfield.hpp"
 #include "roxiemem.hpp"
 
@@ -78,36 +77,6 @@ static int getNumFields(const RtlTypeInfo *record)
     while (*fields++)
         count++;
     return count;
-}
-
-static void handleDeserializeOutcome(DeserializationResult resultcode, const char *targetype, const char *culpritvalue)
-{
-    switch (resultcode)
-    {
-    case Deserialization_SUCCESS:
-        break;
-    case Deserialization_BAD_TYPE:
-        failx("Deserialization error (%s): value cannot be const", targetype);
-        break;
-    case Deserialization_UNSUPPORTED:
-        failx("Deserialization error (%s): encountered value type not supported", targetype);
-        break;
-    case Deserialization_INVALID_TOKEN:
-        failx("Deserialization error (%s): token cannot be NULL, empty, or all whitespace", targetype);
-        break;
-    case Deserialization_NOT_A_NUMBER:
-        failx("Deserialization error (%s): non-numeric characters found in numeric conversion: '%s'", targetype, culpritvalue);
-        break;
-    case Deserialization_OVERFLOW:
-        failx("Deserialization error (%s): number too large to be represented by receiving value", targetype);
-        break;
-    case Deserialization_UNDERFLOW:
-        failx("Deserialization error (%s): number too small to be represented by receiving value", targetype);
-        break;
-    default:
-        typeError(targetype, culpritvalue);
-        break;
-    }
 }
 
 enum PathNodeType {CPNTScalar, CPNTDataset, CPNTSet};
@@ -787,7 +756,7 @@ using TableColumns = std::unordered_map<std::string, std::shared_ptr<arrow::Arra
  * The next function returns the index to read and the table to read from. shouldRead will return true as long as
  * the worker can read another row.
  */
-class ParquetReader
+class PARQUETEMBED_PLUGIN_API ParquetReader
 {
 public:
     ParquetReader(const char *option, const char *_location, int _maxRowCountInTable, const char *_partitionFields, const IThorActivityContext *_activityCtx);
@@ -817,7 +786,7 @@ private:
     std::shared_ptr<arrow::RecordBatchReader> rbatchReader = nullptr;                           // RecordBatchReader reads a dataset one record batch at a time. Must be kept alive for rbatchItr.
     arrow::RecordBatchReader::RecordBatchReaderIterator rbatchItr;                              // Iterator of RecordBatches when reading a partitioned dataset.
     std::vector<__int64> fileTableCounts;                                                       // Count of RowGroups in each open file to get the correct row group when reading specific parts of the file.
-    std::vector<std::unique_ptr<parquet::arrow::FileReader>> parquetFileReaders;                // Vector of FileReaders that match the target file name. data0.parquet, data1.parquet, etc.
+    std::vector<std::shared_ptr<parquet::arrow::FileReader>> parquetFileReaders;                // Vector of FileReaders that match the target file name. data0.parquet, data1.parquet, etc.
     TableColumns parquetTable;                                                                  // The current table being read broken up into columns. Unordered map where the left side is a string of the field name and the right side is an array of the values.
     std::vector<std::string> partitionFields;                                                   // The partitioning schema for reading Directory Partitioned files.
     arrow::MemoryPool *pool = nullptr;                                                          // Memory pool for reading parquet files.
@@ -829,7 +798,7 @@ private:
  * target directory or matching the file mask will be deleted and writing will continue. openWriteFile opens the write file or sets the
  * partitioning options. writeRecordBatch utilizes the open write streams and writes the data to the target location.
  */
-class ParquetWriter
+class PARQUETEMBED_PLUGIN_API ParquetWriter
 {
 public:
     ParquetWriter(const char *option, const char *_destination, int _maxRowCountInBatch, bool _overwrite, arrow::Compression::type _compressionOption, const char *_partitionFields, const IThorActivityContext *_activityCtx);
@@ -896,7 +865,7 @@ private:
  * @brief Builds ECL records for ParquetRowStream.
  *
  */
-class ParquetRowBuilder : public CInterfaceOf<IFieldSource>
+class PARQUETEMBED_PLUGIN_API ParquetRowBuilder : public CInterfaceOf<IFieldSource>
 {
 public:
     ParquetRowBuilder(TableColumns *_resultRows, int64_t _currentRow)
@@ -931,8 +900,6 @@ protected:
 
 private:
     __int64 currentRow;                                                             // The index in the arrow Array to read the current value.
-    TokenDeserializer tokenDeserializer;                                            // Deseralize string type values to numeric types when returning results.
-    TokenSerializer tokenSerializer;                                                // serialize numeric types to string types when returning results.
     StringBuffer serialized;                                                        // Output string from serialization.
     TableColumns *resultRows = nullptr;                                             // A pointer to the result rows map where the left side are the field names for the columns and the right is an array of values.
     std::vector<PathTracker> pathStack;                                             // PathTracker keeps track of nested data when reading sets.
@@ -1005,7 +972,6 @@ protected:
     int firstParam;
     RtlFieldStrInfo dummyField;
     int thisParam;
-    TokenSerializer tokenSerializer;
     std::shared_ptr<ParquetWriter> parquetWriter;
 };
 

